ci/deps.sh

```diff
@@ -34,40 +35,72 @@ echo "MAKEFLAGS: ${MAKEFLAGS}"
 mkdir -p /tmp
 cd /tmp
 
+gh_retrieve_wget(){
+    ORG=$1
+    REPO=$2
+    TAG=$3
+
+    echo "Downloading $ORG $REPO $TAG"
+    
+
+
+    wget $WGET_FLAGS https://github.com/$ORG/$REPO/archive/refs/tags/$TAG.tar.gz && \
+    mkdir $REPO && \
+    tar xzfv $TAG.tar.gz --strip-components=1 -C $REPO && \
+    rm $TAG.tar.gz
+}
+
+gh_retrieve_git(){
+    ORG=$1
+    REPO=$2
+    TAG=$3
+
+    echo "Downloading $ORG $REPO $TAG"
+    git clone --branch $TAG --recursive https://www.github.com/$ORG/$REPO
+}
+
+gh_build(){
+    ORG=$1
+    REPO=$2
+    TAG=$3
+
+    BUILD_CMD=$4
+
+    if [ "${GIT}" ]; then
+        gh_retrieve_git $ORG $REPO $TAG || exit -1
+    else
+        gh_retrieve_wget $ORG $REPO $TAG  || exit -1 
+    fi
+
+    cd $REPO && \
+
+    sh -c "$BUILD_CMD" && \
+    cd .. && \
+    
+    rm -rf $REPO    
+}
+
+
 # Dependencies (not fulfilled by Dockerfile)
-git clone https://github.com/akheron/jansson
-cd jansson ; rm -rf ./.git ; autoreconf -i ; ./configure ; make ; sudo make install ; cd ..
+gh_build "akheron" "jansson" "v2.13.1" "autoreconf -i && ./configure && make && sudo make install" && \
 
-git clone https://github.com/edenhill/librdkafka
-cd librdkafka ; rm -rf ./.git ; ./configure ; make ; sudo make install ; cd ..
+gh_build "edenhill" "librdkafka" "v1.7.0" "./configure && make && sudo make install" && \
 
-wget ${WGET_FLAGS} https://github.com/alanxz/rabbitmq-c/archive/refs/tags/v0.11.0.tar.gz
-mv v0.11.0.tar.gz rabbitmq-c-0.11.0.tar.gz
-tar xfz rabbitmq-c-0.11.0.tar.gz
-cd rabbitmq-c-0.11.0 ; rm -rf ./.git ; mkdir build ; cd build ; cmake -DCMAKE_INSTALL_LIBDIR=lib .. ; sudo cmake --build . --target install ; cd .. ; cd ..
+gh_build "alanxz" "rabbitmq-c" "v0.11.0" "mkdir build && cd build && cmake -DCMAKE_INSTALL_LIBDIR=lib .. && sudo cmake --build . --target install && cd .." && \
 
-git clone --recursive https://github.com/maxmind/libmaxminddb
-cd libmaxminddb ; rm -rf ./.git ; ./bootstrap ; ./configure ; make ; sudo make install ; cd ..
+GIT=1 gh_build "maxmind" "libmaxminddb" "1.6.0" "./bootstrap && ./configure && make && sudo make install" && \
 
-git clone -b 3.4-stable https://github.com/ntop/nDPI
-cd nDPI ; rm -rf ./.git ; ./autogen.sh ; ./configure ; make ; sudo make install ; sudo ldconfig ; cd ..
+GIT=1 gh_build "ntop" "nDPI" "3.4-stable" "./autogen.sh && ./configure && make && sudo make install && sudo ldconfig" && \
```

Comment:
Given that you are building helper functions, but that these are not very large, it might be worth defining `build_git` and `build_wget` instead of passing this env. variable (and in any case I would prefer a parameter than an env. variable in this case)

Reply:
Hi Marc,
Thanks, I'll try to address this at the same time as your comment above.